project/src/js/lichess/friends.ts

```diff
@@ -0,0 +1,88 @@
+class Friend {
+  name: string;
+  playing: boolean;
+
+  constructor(name: string, playing: boolean) {
+    this.name = name;
+    this.playing = playing;
+  }
+}
+
+let onlineFriends: Array <Friend> = [];
+
+function makeFriend(name: string, isPlaying: boolean) {
+  return new Friend(name, isPlaying);
+};
+
+/** Compares usernames for equality, ignoring prefixed titles (such as GM) */
+function isSameUser(userId: string, name: string) {
+  const id = (name.indexOf(' ') >= 0) ? name.split(' ')[1] : name;
+  return id.toLowerCase() === userId;
+};
+
+function findByUsername(n: string) {
+  return onlineFriends.find(u => isSameUser(n.toLowerCase(), u.name));
+}
+
+function setPlaying(userName: string, playing: boolean) {
+  const user = findByUsername(userName);
```

Comment:
Use a lamba here: ```const user = onlineFriends.find(u => ....)```
Don't need `findByUsername` to be a named function, it is used only once.